toolchain/check/dump_id.cpp

```diff
@@ -0,0 +1,51 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/dump_id.h"
+
+#include "common/check.h"
+#include "llvm/Support/raw_ostream.h"
+#include "toolchain/check/context.h"
+#include "toolchain/lex/tokenized_buffer.h"
+#include "toolchain/parse/tree.h"
+#include "toolchain/sem_ir/file.h"
+
+namespace Carbon::Check::DumpIdOverloads {
+
+auto DumpId(SemIR::LocId loc_id, const Context& context) -> void {
+  if (!loc_id.is_valid()) {
+    llvm::errs() << "LocId(invalid)";
+    return;
+  }
+
+  if (loc_id.is_node_id()) {
+    auto token = context.parse_tree().node_token(loc_id.node_id());
+    auto line = context.tokens().GetLineNumber(token);
+    auto col = context.tokens().GetColumnNumber(token);
+    const char* implicit = loc_id.is_implicit() ? " implicit" : "";
+    llvm::errs() << "LocId(line: " << line << ", col: " << col << implicit
```

Comment:
Is it possible to report the filename as well? If so, it might be nice if this were reported in "file:line:col" format (and likewise for the `TokenId` overload). For example, VSCode makes text in that format clickable, which is very handy.

Reply:
Good idea! Done.